### PR TITLE
switch current extension to use python

### PR DESCRIPTION
**Pull request by oussamaessaji**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,120 +1,96 @@
...
-const task = JSON.parse(fs.readFileSync(path.join(__dirname, "task.json")).toString());
-const version = `${task.version.Major}.${task.version.Minor}.${task.version.Patch}`
-
-const javaPath = tl.which("java");
-if (!javaPath) {
-  onError("java is not found in the path", 1);
-}
-const java = tl.tool("java");
-const binOstorlab = path.join(__dirname, "ostorlab_ci.jar");
+async function run_scan(): Promise<void> {
```
Comment (3asm) [inline src/index.ts:32]: Camel case runScan

File: src/index.ts [hunk 2]
```diff
@@ -1,120 +1,96 @@
...
+    try {
 
-java.arg("-jar");
-java.arg(binOstorlab);
+        // Get Python 3 path
+        const pyPath: string = await getPythonPath();
+        console.log('PYTHON PATH: ' + `${pyPath}`);
```
Comment (3asm) [inline src/index.ts:37]:
```suggestion
        debug('PYTHON PATH: ' + `${pyPath}`);
```

File: src/index.ts [hunk 3]
```diff
@@ -1,120 +1,96 @@
...
+            ostorlabExutable.arg(title!)
+            ostorlabExutable.arg("--break-on-risk-rating")
+            ostorlabExutable.arg(riskThreshold)
+            ostorlabExutable.arg("--max-wait-minutes")
+            ostorlabExutable.arg(waitMinutes)
+            ostorlabExutable.arg("--scan-profile")
+            if (scanProfile!= null && scanProfile == "Full Scan") {
+                ostorlabExutable.arg("full_scan")
+            } else {
+                ostorlabExutable.arg("full_scan")
```
Comment (3asm) [inline src/index.ts:74]: full_scan , full_scan!!

File: src/index.ts [hunk 4]
```diff
@@ -1,120 +1,96 @@
...
 
-if (breakBuildOnScore) {
-  java.arg("--breakBuildOnScore");
-  java.arg("--riskThreshold");
-  java.arg(riskThreshold);
-}
+            if (platform!= null && platform.toLowerCase() == "android") {
+                ostorlabExutable.arg("android-apk")
+            } else {
+                ostorlabExutable.arg("ios-ipa")
```
Comment (3asm) [inline src/index.ts:80]: else if, else noimplementederror.

File: src/index.ts [hunk 5]
```diff
@@ -1,120 +1,96 @@
...
 
-java.on("stdout", function (data: Buffer) {
-  const dataValue = data.toString()
+            const result: string = await ostorlabExutable.execSync().stdout;
+            console.log('Scan ID: ' + `${result}`);
+        }
 
-  console.log(`STDOUT: ${dataValue}`)
+    } catch (err: any) { // eslint-disable-line @typescript-eslint/no-explicit-any
+        error(err.message);
```
Comment (3asm) [inline src/index.ts:90]:
```suggestion
        error(`An error was encountered, please check documentation at (https://docs.ostorlab.co) or contact support at support@ostorlab.dev: ${err.message}`);
```

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 6]
```diff
@@ -1,120 +1,150 @@
...
+let credentials: {login : string, pass : string, role : string, url : string}[] = null;
+let customCredentials: {name : string, value : string}[] = null;
 
-if (process.env.SYSTEM_DEBUG) {
-  java.arg("--debug");
+if (extra != null || extra != undefined){
+const extra_dict = YAML.parse(extra);
+  sbomFiles = extra_dict["sbom"]?.toString().split(",");
+  credentials = extra_dict["credentials"];
+  customCredentials = extra_dict["custom_credentials"];
```
Comment (3asm) [inline src/index.ts:41]:
```suggestion
if (extra != null || extra != undefined){
const extraDict = YAML.parse(extra);
  sbomFiles = extraDict["sbom"]?.toString().split(",");
  credentials = extraDict["credentials"];
  customCredentials = extraDict["custom_credentials"];
```

File: vss-extension.json [hunk 7]
```diff
@@ -25,7 +25,7 @@
         "Azure DevOps Extensions",
         "Marketplace"
     ],
-    "public": true,
+    "public": false,
```
Comment (3asm) [inline vss-extension.json:28]: Why is this private?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/task.json [hunk 8]
```diff
@@ -85,15 +82,6 @@
             "helpMarkDown": "Scan title.",
             "groupName": "advanced"
         },
-        {
```
[the remark this answers is not in the document]
Comment (oussamaessaji) [reply src/task.json:88]: The OSS handle this, and the other integrations that use the OSS use the same command.

File: vss-extension.json (hunk 7 above)
Comment (oussamaessaji) [reply vss-extension.json:28]: To be tested, otherwise we need to add another Repo to facilitate the testing

**Review by 3asm — APPROVED** [review]

**Pull request merged**